spec/cb/utils/validator_spec.rb

```diff
@@ -12,127 +12,131 @@
+    let(:response_property) { double('HTTParty::Response.response', body: response_body) }
+    let(:response_body)     { nil }
+    let(:response_code)     { 200 }
 
-    before :each do
-      allow(response).to receive(:response).and_return(response_property)
-      allow(response).to receive(:code).and_return(200)
-    end
+    subject(:validation) { ResponseValidator.validate(response) }
 
-    it 'should return empty hash when body is empty' do
-      allow(response.response).to receive(:body).and_return('')
-      validation = ResponseValidator.validate(response)
-      expect(validation.empty?).to be_truthy
-    end
+    context 'with an empty body' do
+      let(:response_body) { '' }
 
-    it 'should return empty hash when body is improper json/xml' do
-      allow(response.response).to receive(:body).and_return('json')
-      validation = ResponseValidator.validate(response)
-      expect(validation.empty?).to be_truthy
+      it 'returns an empty hash' do
+        expect(validation).to be_empty
+      end
     end
 
-    it 'should return empty hash when response is nil' do
-      response = nil
-      validation = ResponseValidator.validate(response)
-      expect(validation.empty?).to be_truthy
-    end
+    context 'with an invalid body' do
+      let(:response_body) { 'json' }
 
-    it 'should raise DocumentNotFound error if url is invalid' do
-      allow(response.response).to receive(:body).and_return('<!DOCTYPE html></html>')
-      allow(response).to receive(:code).and_return 404
-      expect { ResponseValidator.validate(response) }.to raise_error(Cb::DocumentNotFoundError)
+      it 'returns an empty hash' do
+        expect(validation).to be_empty
+      end
     end
 
-    it 'should return a full json hash when response status is not 200 and content is json' do
-      allow(response.response).to receive(:body).and_return('{"TestJson":{"Test":"True"}}')
-      validation = ResponseValidator.validate(response)
-      expect(validation.empty?).to be_falsey
-    end
+    context 'with a nil body' do
+      let(:response_body) { nil }
 
-    it 'should return a blank json hash when status is not 200 and content is html' do
-      allow(response.response).to receive(:body).and_return('<!DOCTYPE html></html>')
-      validation = ResponseValidator.validate(response)
-      expect(validation.empty?).to be_truthy
+      it 'returns an empty hash' do
+        expect(validation).to be_empty
+      end
     end
 
-    context 'raises a ServiceUnavailableError' do
-      it 'when status code is 503' do
-        allow(response.response).to receive(:body).and_return('{"errors":[]}')
-        allow(response).to receive(:code).and_return 503
-        expect { ResponseValidator.validate(response) }.to raise_error(Cb::ServiceUnavailableError)
+    context 'with an unsuccessful response status' do
+      let(:response_code) { 500 }
```

Comment:
I originally intended to write more specs directly under it but got caught off. I also wanted all of my contexts to explicitly set the case they describe, even if it's overwritten later